### PR TITLE
Remove all the patches removing `ulimit` calls

### PR DESCRIPTION
**Pull request by andreas-kupries**
## Description

Removed all the patches which removed `ulimit` calls from various pre-start and other kubecf scripts.

## Motivation and Context

This is in the context of tickets #504 and #505, where a first set of these patches was given to upstream for consideration. In the recent check of what they were used for the discussion introduced the possibility that the patches might not be necessary anymore, or even if, that the underlying issue could be solved differently, via ops files modifying job capabilities.

Testing without the patches strongly indicated them as not necessary anymore. Thus this PR removes them, and the relevant upstream PRs (https://github.com/cloudfoundry/capi-release/pull/172, https://github.com/cloudfoundry/diego-release/pull/512) will be retracted and closed.

:warning: I am deferring the retractions until after this PR is reviewed, accepted, and merged.

## How Has This Been Tested?

It was tested that both then-master f90161acb9f6bb155e7b70f5638f353b3d803953 and the branch from it without the ulimit patches deploy and then pass the :cat2: tests.

:warning: The tested branch was rebased to the current master commit fef511185b3a0617e8dfb837a86e2e1e4faed8a4, it has not been re-tested.

The initial testing using an SA config on a local minikube with 16/20 G memory failed severely in the :cat2: even for the unchanged baseline, making this attempt useless.

The final testing then happened using a GKE cluster deployed via catapult. The necessary bundle of charts was created from the relevant commit via
> bazel build //deploy/bundle:kubecf-bundle

The catapult configuration used was
```
export AUTOSCALER=false
export BACKEND=gke
export CLUSTER_NAME=_ulimit
export DEBUG_MODE=true
export ENABLE_EIRINI=false
export HA=false
export KUBECF_TEST_SUITE=cats
export SCF_OPERATOR=true
export SCF_TESTGROUP=true
export GKE_CLUSTER_NAME=ak-kubecf-505-ulimit

export SCF_CHART=/work/SUSE/dev/kubecf-1/output/kubecf-bundle.tgz
# ! Modify the above to whatever local location of the bundle

export GKE_CRED_JSON=/work/SUSE/dev/kubecf-1/output/gke-cred.json
# ! Creds pulled out of black box.
```
using the targets
> scf, scf-login, tests-kubecf, scf-clean, clean
to deploy, test and clear the cluster in question.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cc_uploader_ctl.sh [hunk 1]
```diff
@@ -1,27 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/capi/cc_uploader/templates/cc_uploader_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cc_uploader_ctl.sh:5]: As far as I can tell, we can never trigger this file to be used anyway (that is: it's not used with bpm).

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cloud_controller_api_ctl.sh [hunk 2]
```diff
@@ -1,27 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/capi/cloud_controller_ng/templates/cloud_controller_api_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cloud_controller_api_ctl.sh:5]: Same here — this script appears to be unused.

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_file_server_ctl.sh [hunk 3]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/file_server/templates/file_server_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_file_server_ctl.sh:5]: Same.

File: bosh/releases/pre_render_scripts/auctioneer/auctioneer/jobs/patch_auctioneer_ctl.sh [hunk 4]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/auctioneer/templates/auctioneer_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/auctioneer/auctioneer/jobs/patch_auctioneer_ctl.sh:5]: Also same.

File: bosh/releases/pre_render_scripts/diego-api/bbs/jobs/patch_bbs_ctl.sh [hunk 5]
```diff
@@ -1,33 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/bbs/templates/bbs_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/diego-api/bbs/jobs/patch_bbs_ctl.sh:5]: … Yep.

File: bosh/releases/pre_render_scripts/diego-api/locket/jobs/patch_locket_ctl.sh [hunk 6]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/locket/templates/locket_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/diego-api/locket/jobs/patch_locket_ctl.sh:5]: ✅

File: bosh/releases/pre_render_scripts/diego-cell/rep/jobs/patch_rep_ctl.sh [hunk 7]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/rep/templates/rep_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/diego-cell/rep/jobs/patch_rep_ctl.sh:5]: ✅

File: bosh/releases/pre_render_scripts/router/gorouter/jobs/patch_pre-start.sh [hunk 8]
```diff
@@ -1,30 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/routing/gorouter/templates/pre-start.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/router/gorouter/jobs/patch_pre-start.sh:5]: This one actually gets used, I believe. Given that pre-start scripts run in a _different_ container I'm suspicious of it doing anything useful.  Either way, we will need to check the resulting ulimit… I see that the `gorouter` binary has an actual limit of `1048576` (for both soft and hard), so we're _not_ pulling the value from `bpm.yml`, interesting.

File: bosh/releases/pre_render_scripts/scheduler/ssh_proxy/jobs/patch_ssh_proxy_ctl.sh [hunk 9]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
-
-set -o errexit -o nounset
-
-target="/var/vcap/all-releases/jobs-src/diego/ssh_proxy/templates/ssh_proxy_ctl.erb"
```
Comment (mook-as) [inline bosh/releases/pre_render_scripts/scheduler/ssh_proxy/jobs/patch_ssh_proxy_ctl.sh:5]: Appears to be unused.

**Comment by andreas-kupries** [comment]
Thank you for the reminder about the commit message. Will fix.

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cc_uploader_ctl.sh (hunk 1 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cc_uploader_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_file_server_ctl.sh (hunk 3 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_file_server_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/diego-api/bbs/jobs/patch_bbs_ctl.sh (hunk 5 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/diego-api/bbs/jobs/patch_bbs_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/diego-api/locket/jobs/patch_locket_ctl.sh (hunk 6 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/diego-api/locket/jobs/patch_locket_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/diego-cell/rep/jobs/patch_rep_ctl.sh (hunk 7 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/diego-cell/rep/jobs/patch_rep_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/router/gorouter/jobs/patch_pre-start.sh (hunk 8 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/router/gorouter/jobs/patch_pre-start.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/scheduler/ssh_proxy/jobs/patch_ssh_proxy_ctl.sh (hunk 9 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/scheduler/ssh_proxy/jobs/patch_ssh_proxy_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/auctioneer/auctioneer/jobs/patch_auctioneer_ctl.sh (hunk 4 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/auctioneer/auctioneer/jobs/patch_auctioneer_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

File: bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cloud_controller_api_ctl.sh (hunk 2 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cloud_controller_api_ctl.sh:5]: Squashed history and amended commit message based on these comments.
Commit 5f7cd28c48738106936b93d1ae215ead9c8d412d

**Comment by andreas-kupries** [comment]
Squashed history and amended the commit message based on Mark's comments.
Rebased.

**Pull request merged**